src/modelize/modelize_property.nit

```diff
@@ -874,9 +856,10 @@ redef class AMethPropdef
 		mclassdef.mprop2npropdef[mprop] = self
 
 		var mpropdef = new MMethodDef(mclassdef, mprop, self.location)
-		if mprop.name == "autoinit" and mclassdef.is_intro then
-			assert mclassdef.auto_init == null
-			mclassdef.auto_init = mpropdef
+		if mpropdef.name == "defaultinit" and mclassdef.is_intro then
+			assert mclassdef.default_init == null
+			mpropdef.is_old_style_init = is_old_style_init
```

Comment:
why 'is_old_style_init'

Reply:
When you declar an explicit old_style_init `init(a: A) is old_style_init` this property replaces the default_init so we have now the old_style_init as the default constructor. It is the purpose of this flag, to determine if the method is a auto-generated default_init or if it has been specified by the user with the is_old_style_init annotation.